### PR TITLE
Simplify/correct definition links

### PR DESCRIPTION
**Pull request by patrickhlauke**
Closes https://github.com/w3c/wcag/issues/3920

### CODE REVIEW

**Comment by kfranqueiro** [comment]
This looks good to me for techniques and understanding. I would suggest removing the changes in this PR for conformance-challenges and guidelines since those aren't processed the same way. Sorry I was probably ambiguous about that in the issue.

**Comment by patrickhlauke** [comment]
Reverted those two non-technique/non-understanding changes

**Comment by kfranqueiro** [comment]
**TL;DR: we should not merge this until we're using the Eleventy build process, where I've fixed an underlying issue.**

This isn't so much a problem with this PR as it is with the XSLT build process. In the process of thoroughly testing, I think I understand why some of these (particularly some under `techniques`) were using full URLs.

The XSLT build process generates its own IDs for definitions links, disregarding the already-existing `id` attributes in the markup in `guidelines/terms/*` files. This is fine for Understanding pages, which cross-links to headings under its own Key Terms section. However, Techniques pages don't do this; they link directly to the TR - which is bad if the IDs don't match.

As it stands, the links to "status messages", "change of context", "conforming alternate version", and "web pages" will not scroll to the correct position in the TR when this PR's changes are processed by the XSLT build process. I have added a commit to the Eleventy build process PR (#3917) to address this for Techniques pages, so this PR will work 100% with that build process.

**Review by kfranqueiro — APPROVED** [review]
Tthe Eleventy build process has now been successfully used to push the latest site update, and merged into the main branch, so it's now safe to merge this PR.

**Review by stevefaulkner — APPROVED** [review]

File: techniques/failures/F61.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
                     automatic update that the user cannot disable from within the content</h1><section class="meta"><p class="id">ID: F61</p><p class="technology">Technology: failures</p><p class="type">Type: Failure</p></section><section id="applicability"><h2>When to Use</h2>
       <p>General </p>
    </section><section id="description"><h2>Description</h2>
-      <p> This document describes a failure that occurs when the content in the main <a href="/TR/WCAG/#dfn-viewport">viewport</a> viewport is automatically updated, and there is no option for a user to disable this behavior. </p>
+      <p> This document describes a failure that occurs when the content in the main <a>viewport</a> viewport is automatically updated, and there is no option for a user to disable this behavior. </p>
```
Comment (mbgower) [inline techniques/failures/F61.html:6]: I'm assuming this is only supposed to be the word "viewport" once?
```suggestion
      <p> This document describes a failure that occurs when the content in the main <a>viewport</a> is automatically updated, and there is no option for a user to disable this behavior. </p>
```
